projects/js-packages/charts/src/providers/chart-context/global-charts-provider.tsx

```diff
@@ -161,15 +161,16 @@ export const GlobalChartsProvider: FC< GlobalChartsProviderProps > = ( { childre
 					return existing;
 				}
 
+				// Use map size as index to assign colors sequentially (0, 1, 2...)
+				// ensuring each new group gets the next available palette color
 				const assignedCount = groupToColorMap.size;
-				const color =
-					colorCache.colors.length > 0 ? getChartColor( assignedCount, colorCache ) : '#000000';
+				const color = getChartColor( assignedCount, colorCache );
 				groupToColorMap.set( group, color );
 
 				return color;
 			}
 
-			return colorCache.colors.length > 0 ? getChartColor( index, colorCache ) : '#000000';
+			return getChartColor( index, colorCache );
```

Comment:
This change prevents an issue I observed in Storybook where the country colors were all black.

I believe this was a mistake in my original implementation. There is no reason to return a fallback color here. If the color cache is empty we can just generate a color based on the theme colors.